ONSdigital/eq-author-app
# EAR 833 - Improve unit selector

### What is the context of this PR?

This PR involves changing the unit selector for unit answer types from a `<select>` to an new `autocomplete` component.

I've added the accessibility addon for storybook to author to help with accessibility going forward

I've also added `userEvent` to `react-testing-library` to get natural dom events.

Quick run down, `keyDown` events weren't causing the input focus to change in the `rtl` tests. They do with `userEvent`. Seems pretty straightforward and more about it is here:
https://github.com/testing-library/user-event

### How to review

1. Create a questionnaire
2. Make a unit type
3. Select a type by either using `space`, `enter` or `mouse click`
4. Check it saves in the JSON
5. Remove it and check it returns to `""`;
6. Also that validation errors reappear
